File: analysisPass/analysis.py
# モジュールのインポート
import time 
import getpass
from itertools import product 

target = getpass.getpass(prompt='パスワードを入力してください: ')# パスワードを入力

chars = '0123456789abcdefghijklmnopqrstuvwxyz'# すべての文字

# パスワード解析関数
def check(text,repeat):
    passwords = product(text,repeat=repeat) # 与えられた文字から生成可能な全てのパスワードを生成
    for i, password in enumerate(passwords): # 各パスワードをfor文で総なめ
        print('\033[32m'+str(i)+' 回目の処理 : '+''.join(password)+'\033[0m') #実行回数と生成した候補パスワードを表示
        if ''.join(password)==target: # 生成したパスワードが目標のパスワードと一致するか確認
            return ''.join(password) #一致した場合のみパスワードをreturn 
        
# 解析実行判断関数
def yes_no_input():
    while True:
        choice = input('パスワードの解析を始めますか? [yes/no]: ').lower()
        if choice == 'yes':# yesなら解析実行
            return True
        else:
            return False
        

if yes_no_input():
    start = time.time() # 時間(s)計測開始
    pw = check(chars,3) # パスワード解析関数の呼び出し 第二引数は解析する文字数
    if pw is None:
        print('失敗')
    else:
        print()
        print('パスワードが見つかりました ----->>>',pw)
        finish = time.time()-start #　解析時間(s)の計算
        print(finish,'秒' ,'=', finish/60,'分')
else:
    print('program end.')

File: analysisPass/analysis.java

import java.util.*;
import java.io.Console;

public class analysis {
    static String target;
    static String chars = "0123456789abcdefghijklmnopqrstuvwxyz";
    static int maxLen = 4;

    public static void main(String[] args) {
        Console console = System.console();
        if (console == null) {
            System.out.println("Couldn't get Console instance");
            System.exit(0);
        }

        char[] passwordArray = console.readPassword("パスワードを入力してください: ");
        target = new String(passwordArray);

        if (yesNoInput()) {
            long start = System.currentTimeMillis();
            String pw = check("", 3);
            if (pw == null) {
                System.out.println("失敗");
            } else {
                System.out.println();
                System.out.println("パスワードが見つかりました ----->>> " + pw);
                long finish = System.currentTimeMillis() - start;
                System.out.println(finish / 1000.0 + "秒 = " + finish / 60000.0 + "分");
            }
        } else {
            System.out.println("program end.");
        }
    }

    static String check(String curr, int len) {
        if (len == maxLen) {
            System.out.println(curr + " 回目の処理 : " + curr);
            if (curr.equals(target)) {
                return curr;
            } else {
                return null;
            }
        } else {
            for (char c : chars.toCharArray()) {
                String result = check(curr + c, len + 1);
                if (result != null) {
                    return result;
                }
            }
            return null;
        }
    }

    static boolean yesNoInput() {
        Scanner scanner = new Scanner(System.in);
        while (true) {
            System.out.print("パスワードの解析を始めますか? [yes/no]: ");
            String choice = scanner.nextLine().toLowerCase();
            if (choice.equals("yes")) {
                return true;
            } else {
                return false;
            }
        }
    }
}
